assimilation_code/modules/io/direct_netcdf_mod.f90

```diff
@@ -3030,17 +3064,34 @@ subroutine set_dart_missing_value(array, domain, variable)
 
 endif
 
+if ( get_has_FillValue(domain, variable) ) then 
+
+   select case ( get_xtype(domain, variable) )
+      case ( NF90_INT )
+         call get_FillValue(domain, variable, model_missing_valueINT)
+         where(array == model_missing_valueINT) array = MISSING_R8
```

Comment:
This logic of letting the compiler do the variable type conversion mimics how the variables are read. Since we are testing for direct equality, I wanted to make sure `array` and `model_missing_xxxx` are type-converted the same.

Reply:
i think this code is doing more work than is needed and is confusing.  we have already read the netcdf data into an R8 (or R4) array and that has done whatever conversion/roundoff might be needed (e.g. int -> real_r8 or float (real_r4) -> real_r8).  why not read the fill value directly into an R8 variable which will do the same conversion as the array.  then you don't need the different variable types, nor the case statement and you can do a single where() statement.

Reply:
CLM has both NF90_DOUBLE and NF90_FLOAT  variables, both with attributes for missing_value and _FillValue.
If the variable is NF90_DOUBLE, the fill value reported by ncdump is 'missing_value = 1.e+36 ;`
if the variable is NF90_FLOAT, the fill value is `missing_value = 1.e+36f ;`
so there is some information that the attribute has a matching netCDF 'xtype' if you will. 

I have not convinced myself that the conversion of the attribute value is the same as the conversion of the variable values. The variable values have a netCDF 'xtype', it is not known to me if the attribute values are of the same netCDF 'xtype' and are converted to r8 (or r4) identically. Do you know?

When we go to replace the DART missing value in the state with the (one) CLM special value (we've checked that if both are specified that they are identical), I want to make sure we put back EXACTLY what we found. The existing netCDF files may have NF90_DOUBLE variables and we might be running in reduced precision ... so we have to convert to the precision of the existing output variable,
using that exact missing value. So - I was keeping both in the state structure, even if we run with r8=r4.